src/main/java/edu/umich/tl/CourseDelete.java

```diff
@@ -62,32 +93,200 @@ private static void checkForApiType(Properties properties) {
-			M_log.error("The api call getting \"Enrollemnt terms\" is UnSuccessfull has IllegalStateException ",e);
+		} catch (JsonParseException e) {
+			M_log.error("getPreviousTerms() has JsonParseException",e);
+		} catch (JsonMappingException e) {
+			M_log.error("getPreviousTerms() has JsonMappingException",e);
 		} catch (IOException e) {
-			M_log.error("The api call getting \"Enrollemnt terms\" is UnSuccessfull has IOException ",e);
+			M_log.error("getPreviousTerms() has IOException",e);
+		}catch (ParseException e) {
+			M_log.error("getPreviousTerms() has ParseException",e);
+		}
+	}
+    /*
+     * For Provided sis_term_id from the properties file 'canvasCourseDelete.properties' we try to
+     * get corresponding canvasTermId for making further APi call
+     */
+	private static String getCanvasTermIdForSisTermId(CoursesForDelete courseForDelete) {
+		M_log.debug("getCanvasTermIdForSisTermId(): called");
+		ArrayList<Term> terms = courseForDelete.getTerms();
+		for (Term term : terms) {
+			if (term.getSisTermId().equalsIgnoreCase(sisTermIdFromPropsFile)) {
+				M_log.info("Found canvasTermId for provided SISTermId for the Term:  "+ term.getTermName());
+				return term.getCanvasTermId();
+			}
+		}
+		return null;
+	}
+	/*
+	 * Getting all the unpublished course list for a particular term from the ROOT account which is '1'. All the colleges, school of UOfM fall under ROOT.
+	 * This list is huge, canvas only provided 100 result set per api call so we need to get the rest of the courses info from the Pagination links.
+	 */
+	private static void getUnpublishedCourseList(String canvasTermIdForSisTermId, CoursesForDelete coursesForDelete, ApiCallHandler apiHandler, String url) {
+		M_log.debug("getUnpublishedCourseList(): Called");
+		HttpResponse httpResponse=null;
+		if(url!=null) {
+			//sending null for canvasTermId is fine since we get the url framed from the pagination object from the response header. 
+			httpResponse=apiHandler.getApiResponse(CanvasApiEnum.UNPUBLISHED_COURSE_LIST_PAGINATION_URL,null,url);
+		}else {
+			httpResponse = apiHandler.getApiResponse(CanvasApiEnum.UNPUBLISHED_COURSE_LIST,canvasTermIdForSisTermId,url);
+		}
+		ObjectMapper mapper = new ObjectMapper();
+		int statusCode = httpResponse.getStatusLine().getStatusCode();
+		if(statusCode!=200) {
+			M_log.error(apiCallErrorHandler(httpResponse,"Unpublished_Courses",apiHandler));
+			System.exit(0);
+		}
+		HttpEntity entity = httpResponse.getEntity();
+		List<HashMap<String, Object>> courseList=new ArrayList<HashMap<String, Object>>();
+		try {
+			String jsonResponseString = EntityUtils.toString(entity);
+			courseList = mapper.readValue(jsonResponseString,new TypeReference<List<Object>>(){});
+			for (HashMap<String, Object> course : courseList) {
+				Course aCourse=new Course();
+				aCourse.setId((Integer)course.get("id"));
+				aCourse.setCourseName((String)course.get("name"));
+				aCourse.setStartDate((String)course.get("start_at"));
```

Comment:
The similar line for setting the start date of terms uses a date type instead of a string.  If the same were done here, it would make the code safer, because the date strings would need to be validated in order to turn them into date objects.